app/src/org/commcare/android/resource/installers/ProfileAndroidInstaller.java

```diff
@@ -93,9 +99,22 @@ public boolean install(Resource r, ResourceLocation location, Reference ref, Res
                     upgrade ? Resource.RESOURCE_STATUS_UNINITIALIZED : Resource.RESOURCE_STATUS_UNINITIALIZED, false);
             
             Profile p = parser.parse();
+            instance.setProfile(p);
             
             if(!upgrade) {
                 initProperties(p);
+
+                // Check that this app is not already installed on the phone
+                String newAppId = p.getUniqueId();
+                ArrayList<ApplicationRecord> installedApps = CommCareApplication._().
```

Comment:
Very minor feedback: I'd split the portion of this that actually runs through the application records and compares the ID into its own method, since that portion is what introduces the CommCareApplication static state dependence.